Source/WebCore/platform/graphics/skia/GraphicsContextSkia.cpp

```diff
@@ -480,11 +498,22 @@ SkPaint GraphicsContextSkia::createStrokePaint() const
 void GraphicsContextSkia::setupStrokeSource(SkPaint& paint) const
 {
     if (auto strokePattern = strokeBrush().pattern())
-        paint.setShader(strokePattern->createPlatformPattern({ }, toSkSamplingOptions(imageInterpolationQuality())));
+        setupShader(paint, strokePattern->createPlatformPattern({ }, toSkSamplingOptions(imageInterpolationQuality())));
     else if (auto strokeGradient = strokeBrush().gradient())
-        paint.setShader(strokeGradient->shader(alpha(), strokeBrush().gradientSpaceTransform()));
+        setupShader(paint, strokeGradient->shader(alpha(), strokeBrush().gradientSpaceTransform()));
     else
-        paint.setColor(SkColor(strokeBrush().color().colorWithAlphaMultipliedBy(alpha())));
+        setupColor(paint, SkColor(strokeBrush().color().colorWithAlphaMultipliedBy(alpha())));
+}
+
+void GraphicsContextSkia::setupColor(SkPaint& paint, SkColor color) const
+{
+    paint.setColor4f(SkColor4f::FromColor(color), colorSpace().platformColorSpace().get());
+}
```

Comment:
I think it's better to do this in every caller, it will be a long line, but you don't have to go to the helper function to know what setting up color means

Reply:
hmm, I'm, not sure I understand this correctly either, I think the color space parameter is to specify the color space of the color we are setting, not the destination color space. Or am I misunderstanding what `colorSpace()` is?

Reply:
You're correct. And since `SkPaint` assumes colors without specified color space are in `sRGB`, we need to make sure the color will be interpreted as if it was in destination color space so that no color conversions are done.